lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -974,13 +1021,16 @@ internal class EvaluatingCompiler(
                                 // Create a closure that groups all the rows in the FROM source into a single group.
                                 thunkFactory.thunkEnv(metas) { env ->
                                     // Evaluate the FROM clause
-                                    val fromProductions: Sequence<FromProduction> = sourceThunks(env)
+                                    var fromProductions: Sequence<FromProduction> = sourceThunks(env)
                                     val registers = createRegisterBank()
 
                                     // note: the group key can be anything here because we only ever have a single
                                     // group when aggregates are used without GROUP BY expression
                                     val syntheticGroup = Group(valueFactory.nullValue, registers)
 
+                                    if (limit != null) {
+                                        fromProductions = fromProductions.take(compileLimit(limit, env))
+                                    }
```

Comment:
Can it be done using `let` to avoid using `var`?

Reply:
+1

Reply:
Updated